CombinatorialGames/Game/Concrete.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2025 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import CombinatorialGames.Game.Impartial
+
+/-!
+# Combinatorial games from a type of states
+
+A "concrete" game is a type of states endowed with well-founded subsequence relations for the
+left and right players. This is often a more convenient representation for a game, which can then be
+used to define a `PGame` or any of the other combinatorial game types.
+-/
+
+open PGame
+
+variable {α : Type*}
+
+/-- A "concrete" game is a type of states endowed with well-founded subsequence relations for the
+left and right players. -/
+class ConcreteGame (α : Type*) where
+  /-- The subsequence relation for the left player. -/
+  subsequentL : α → α → Prop
+  /-- The subsequence relation for the right player. -/
+  subsequentR : α → α → Prop
+  /-- The subsequence relation is well-founded. -/
+  isWellFounded_subsequent : IsWellFounded α fun a b ↦ subsequentL a b ∨ subsequentR a b
+
+namespace ConcreteGame
+variable [ConcreteGame α]
+
+scoped infix:50 " ≺ₗ " => subsequentL
+scoped infix:50 " ≺ᵣ " => subsequentR
+attribute [instance] isWellFounded_subsequent
+
+theorem subrelation_subsequentL :
+    Subrelation subsequentL fun a b : α ↦ subsequentL a b ∨ subsequentR a b :=
+  Or.inl
+theorem subrelation_subsequentR :
```

Comment:
I am generally made uncomftorable by the difference in newline styling compared to mathlib4 (using zero newlines for related theorems and one newline for slightly related theorems compared to one newline and two newlines respectfully), though I suppose that is okay.

Reply:
I've seen some limited usage of "theorems without whitespace in between" in Mathlib, though it's often with one-liners and maybe I'm pushing the envelope a bit too hard. I'll space out these longer theorems.